libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -343,59 +344,48 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
    */
   private _disallowInsecureSha1(
     customTipsConfiguration: CustomTipsConfiguration,
+    exmeptPackageVersions: Record<string, Array<string>>,
     terminal: ITerminal
   ): boolean {
-    const { packages } = this._shrinkwrapJson;
-    if (packages) {
-      for (const { resolution } of Object.values(packages)) {
-        if (resolution?.integrity.startsWith('sha1')) {
-          terminal.writeErrorLine(
-            'Error: An integrity field with "sha1" was found in pnpm-lock.yaml;' +
-              ' this conflicts with the "disallowInsecureSha1" policy from pnpm-config.json.\n'
-          );
+    const exmeptPackageList: Map<string, boolean> = new Map();
+    for (const [pkgName, versions] of Object.entries(exmeptPackageVersions)) {
+      versions.forEach((version) => exmeptPackageList.set(this._getPackageId(pkgName, version), true));
```

Comment:
use `for (const version of versions` to avoid an unnecessary closure and make it easier to step through the loop in a debugger.